netlify/functions/manage-findings.ts

```diff
@@ -403,47 +405,69 @@ async function editFinding(
       emailBody;
     edited = true;
   } else if (
-    !isQaOrGasSubmission &&
-    data.isMitigated?.oldValue !== data.isMitigated?.newValue &&
-    data.isMitigated?.newValue
+    data.mitigationStatus?.oldValue !== data.mitigationStatus?.newValue
   ) {
-    try {
-      await client.request(
-        "DELETE /repos/{owner}/{repo}/issues/{issue_number}/labels/{name}",
-        {
-          owner: process.env.GITHUB_REPO_OWNER!,
-          repo: repoName,
-          issue_number: issueNumber,
-          name: data.risk.oldValue,
-        }
-      );
-    } catch (error) {
-      throw {
-        status: error.status || 500,
-        message: `Error removing old risk label: [${data.risk.oldValue}] - ${
-          error.message || "unknown"
-        }`,
-      };
+    let labelNameToBeRemoved = "";
+    if (
+      data.mitigationStatus?.oldValue === MitigationStatus.MitigationConfirmed
+    ) {
+      labelNameToBeRemoved = "mitigation-confirmed";
+    } else if (
+      data.mitigationStatus?.oldValue === MitigationStatus.Unmitigated
+    ) {
+      labelNameToBeRemoved = "unmitigated";
+    }
+    if (labelNameToBeRemoved) {
+      try {
+        await client.request(
+          "DELETE /repos/{owner}/{repo}/issues/{issue_number}/labels/{name}",
+          {
+            owner: process.env.GITHUB_REPO_OWNER!,
+            repo: repoName,
+            issue_number: issueNumber,
+            name: labelNameToBeRemoved,
+          }
+        );
+      } catch (error) {
+        throw {
+          status: error.status || 500,
+          message: `Error removing old label: [${labelNameToBeRemoved}] - ${
+            error.message || "unknown"
+          }`,
+        };
+      }
     }
 
-    const mitigationConfirmedLabel = "mitigation-confirmed";
-    try {
-      await client.request(
-        "POST /repos/{owner}/{repo}/issues/{issue_number}/labels",
-        {
-          owner: process.env.GITHUB_REPO_OWNER!,
-          repo: repoName,
-          issue_number: issueNumber,
-          labels: [mitigationConfirmedLabel],
-        }
-      );
-    } catch (error) {
-      throw {
-        status: error.status || 500,
-        message: `Error adding new label: [mitigation-confirmed] - ${
-          error.message || "unknown"
-        }`,
-      };
+    let labelNameToAdd = "";
+    if (
+      data.mitigationStatus?.newValue === MitigationStatus.MitigationConfirmed
+    ) {
+      labelNameToAdd = "mitigation-confirmed";
+    } else if (
+      data.mitigationStatus?.newValue === MitigationStatus.Unmitigated
+    ) {
+      labelNameToAdd = "unmitigated";
+    }
```

Comment:
could you add this logic in the lines above here (410-419) where you are specifying what label to remove you could also add the label to add this this is a type of toggle just to make this a little more concise and readable? Not hindering anything but just a suggestion.